.github/actions/detect-workflow/main_test.go

```diff
@@ -172,3 +174,67 @@ func Test_action_getRepoRef(t *testing.T) {
 		}
 	})
 }
+
+func Test_newAction(t *testing.T) {
+	type args struct {
+		getenv func(string) string
+		c      *github.OIDCClient
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *action
+		wantErr bool
+	}{
+		{
+			name: "failure with empty string",
+			args: args{
+				getenv: func(k string) string {
+					return ""
+				},
+			},
+			wantErr: true,
+			want:    nil,
+		},
+		{
+			name: "success",
+			args: args{
+				getenv: func(k string) string {
+					// create a temp file with key:value
+					f, err := ioutil.TempFile("", "")
```

Comment:
use `os.CreateTemp`

Reply:
Is there a specific reason for the request for change?  `ioutil.TempFile` does create a `temp file`

A code review should provide the reason and justification. https://google.github.io/eng-practices/review/reviewer/comments.html#why

The "use os.CreateTemp" doesn't come out as a request, especially for someone who is trying to contribute. https://google.github.io/eng-practices/review/reviewer/comments.html#courtesy

Reply:
Apologies. I would like this to be `os.CreateTemp`. The reason is that `ioutil.TempFile` just calls `os.CreateTemp` anyway as of Go 1.17 (see [godoc](https://pkg.go.dev/io/ioutil#TempFile)). Use of the `ioutil` package seems generally discouraged in the Go community.

Reply:
Thank you! Appreciate the explanation. I will change it.

Reply:
np. Thanks for following up!

Reply:
Also, I commented on this PR in a bit of haste. It was not my intention to be discourteous. Sorry about that! I'll try to explain my thoughts in more detail next time.